jgrapht-core/src/test/java/org/jgrapht/alg/linkprediction/AdamicAdarIndexLinkPredictionTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * (C) Copyright 2020-2020, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.linkprediction;
+
+import static org.junit.Assert.assertArrayEquals;
+
+import org.jgrapht.Graph;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.builder.GraphTypeBuilder;
+import org.jgrapht.util.SupplierUtil;
+import org.junit.Test;
+
+/**
+ * Tests
+ *
+ * @author Dimitrios Michail
+ */
+public class AdamicAdarIndexLinkPredictionTest
+{
+
+    @Test
+    public void testPrediction()
+    {
+        Graph<Integer,
+            DefaultEdge> g = GraphTypeBuilder
+                .undirected().weighted(false).vertexSupplier(SupplierUtil.createIntegerSupplier())
+                .edgeSupplier(SupplierUtil.DEFAULT_EDGE_SUPPLIER).buildGraph();
+
+        for (int i = 0; i < 6; i++)
+            g.addVertex(i);
+
+        g.addEdge(0, 1);
```

Comment:
It's up to you, but there's another way to construct a graph using `TestUtil`, which avoids manually adding edges and vertices (which's done for every test case).